### ISSUE TITLE
Can't Connect to Containers when Firewall blocks STUN traffic

### ISSUE THREAD

**Issue by ryanshaut** [kasmtech/workspaces-issues#294]
Issue: When the Kasm host VM is blocked from internet STUN servers, connection to provisioned containers fails. After creating a container, the page presents 'Creating secure connection', then redirects to another page (if you are signed in as an admin, it redirects to the admin homepage)

![image](https://user-images.githubusercontent.com/4265693/205965502-ea3ee442-54f3-433a-b738-e90b6a5f139a.png)

The Chrome devtools show a 502 to 
https://hostname.example.com/desktop/707bf3a4-e0ae-41f1-bbec-6d24c71d2033/vnc/vnc.html?video_quality=2&enable_webp=false&idle_disconnect=false&password=&autoconnect=1&path=desktop/707bf3a4-e0ae-41f1-bbec-6d24c71d2033/vnc/websockify&cursor=true&resize=remote&clipboard_up=true&clipboard_down=true&clipboard_seamless=true&toggle_control_panel=false

Error logs of the created application container

     2022-12-05 16:39:42,780 [INFO] ICE: Querying public IP...
     2022-12-05 16:39:42,780 [DEBUG] ICE: Trying 'stun.l.google.com', port 19302
     2022-12-05 16:39:52,787 [INFO] ICE: STUN server 0 didn't work, trying next...
     2022-12-05 16:39:52,787 [DEBUG] ICE: Trying 'stun1.l.google.com', port 19302
     2022-12-05 16:40:02,799 [INFO] ICE: STUN server 1 didn't work, trying next...
     2022-12-05 16:40:02,799 [DEBUG] ICE: Trying 'stun2.l.google.com', port 19302
     2022-12-05 16:40:12,812 [INFO] ICE: STUN server 2 didn't work, trying next...
     2022-12-05 16:40:12,812 [DEBUG] ICE: Trying 'stun3.l.google.com', port 19302
     2022-12-05 16:40:22,824 [INFO] ICE: STUN server 3 didn't work, trying next...
     2022-12-05 16:40:22,824 [DEBUG] ICE: Trying 'stun4.l.google.com', port 19302
     2022-12-05 16:40:32,833 [INFO] ICE: STUN server 4 didn't work, trying next...
     2022-12-05 16:40:32,834 [DEBUG] ICE: Trying 'stun.voipbuster.com', port 3478
     2022-12-05 16:40:42,845 [INFO] ICE: STUN server 5 didn't work, trying next...
     2022-12-05 16:40:42,846 [DEBUG] ICE: Trying 'stun.voipstunt.com', port 3478
     2022-12-05 16:40:52,850 [INFO] ICE: STUN server 6 didn't work, trying next...
     2022-12-05 16:40:52,850 [PRIO] ICE: Failed to get public IP, please specify it with -publicIP

Environment:
Kasm installed per doc instructions https://kasmweb.com/docs/latest/install/single_server_install.html#installation-guide
Browser: Chrome Version 108.0.5359.94 (Official Build) (64-bit) on Windows 11.
Host: Debian 11 hosted on Proxmox 7
Network: Host is secured behind Pfsense, blocking most non HTTP/S traffic

Solution:
Add outbound firewall rules to allow traffic to the above hosts and UDP ports.

**Comment by j-travis** [comment]
Thanks for reporting.
I believe you should be able to disable this behavior by adding the `VNC_OPTIONS` environment variable to the Docker Run Config Override.

Here is an example
```
{
  "hostname": "kasm",
  "environment": {
    "VNCOPTIONS": "-publicIP 127.0.0.1"
  }
}
```
![image](https://user-images.githubusercontent.com/48762190/205987037-935c0758-91ba-4b17-b885-6c56d8d0903d.png)
 
 Can you verify this works for you?

**Comment by ryanshaut** [comment]
Yes, that exact config (with 127.0.0.1 IP) worked. Is that only configurable each workspace. Is there a global Docker Run Config Override setting somewhere?

**Comment by j-travis** [comment]
There is a Group Setting named `run_config`  that accepts the same format. It will be used for all sessions for those users.

You can learn more about group settings here:

- https://kasmweb.com/docs/latest/guide/groups/group_settings.html
- https://www.youtube.com/watch?v=teJSUxKEbfE&t=105s

**Comment by luc-vocab** [comment]
I had to apply the above docker config override, however I can control my firewall rules, would I benefit from better performance if I somehow made this STUN protocol work ? Right now I only allow http/https in.

**Comment by j-travis** [comment]
This issue is resolved my default in the new `1.13.0` images eg `kasmweb/chrome:1.13.0`
